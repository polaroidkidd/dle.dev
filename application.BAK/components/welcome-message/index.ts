import { WelcomeMessage } from "./welcome-message";

export { WelcomeMessage };