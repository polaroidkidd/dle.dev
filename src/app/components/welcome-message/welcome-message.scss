@import "../../styles/main";

.welcome-message {
  display: flex;
  justify-content: center;
  text-align: center;
  @media (max-width: 768px) {
    font-size: 2.5rem;
    font-weight: 200;
  }
}

.additional-paragraph-styles {
  margin-top: 1rem;
  text-align: center;
}