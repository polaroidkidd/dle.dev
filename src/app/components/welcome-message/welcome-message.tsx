import React from "react";
import "./welcome-message.scss";

type Props = {
  heading: string
  message?: string
}

const WelcomeMessage: React.FC<Props> = ({heading, message}) => {
  return (
    <>
      <section>
        <h2 className="heading-secondary welcome-message">{heading}</h2>
        {
          message ? <p className="paragraph additional-paragraph-styles">{message} </p> : ""
        }
      </section>
    </>
  );
};
export { WelcomeMessage };