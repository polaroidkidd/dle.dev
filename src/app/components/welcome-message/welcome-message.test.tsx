import React from "react";
import renderer from "react-test-renderer";
import { WelcomeMessage } from "./welcome-message";

test("Render WelcomeMessage with text", () => {
  const tree = renderer.create(<WelcomeMessage heading="test heading"/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Render WelcomeMessage with empty string", () => {
  const tree = renderer.create(<WelcomeMessage heading=""/>).toJSON();
  expect(tree).toMatchSnapshot();
});
