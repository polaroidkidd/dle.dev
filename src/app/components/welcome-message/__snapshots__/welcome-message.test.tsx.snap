// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render WelcomeMessage with empty string 1`] = `
<section>
  <h2
    className="heading-secondary welcome-message"
  >
    
  </h2>
  
</section>
`;

exports[`Render WelcomeMessage with text 1`] = `
<section>
  <h2
    className="heading-secondary welcome-message"
  >
    test heading
  </h2>
  
</section>
`;
